### PR TITLE
[16.0][MIG] - product_price_category

### PR DESCRIPTION
**Pull request by sbejaoui**

### CODE REVIEW

File: product_price_category/views/product_template.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<odoo>
+    <!-- Adds price_category_id in form view -->
+    <record id="product_template_form_view" model="ir.ui.view">
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_form_view" />
+        <field name="arch" type="xml">
+            <field name="list_price" position="after">
+                <field name="price_category_id" />
```
Comment (flotho) [inline product_price_category/views/product_template.xml:8]: Small comment here,
When there is no price category, the field is not visible :+1: 
image

Any chance to add a label?

**Review by flotho — CHANGES_REQUESTED** [review]
small changes requested on view, otherwise LGTM

File: product_price_category/views/product_template.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            </field>
```
Comment (lmignon) [inline product_price_category/views/product_template.xml]:
```suggestion
            <div name="standard_price_uom" position="after">
                <field name="price_category_id" />
            </div>
```

**Review by lmignon — CHANGES_REQUESTED** [review]
@sbejaoui LGTM, just a little change required to fix the UI (Code review + functional tests)

**Comment by rousseldenis** [comment]
@sbejaoui Don't forget to do pre-commit/formatting stuffs in another commit in order to have migration related stuffs only in migration commit. Follow https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0

(pre-commit stuff commit with --no-verify command)

File: product_price_category/models/product_price_category.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2016 Camptocamp SA
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductPriceCategory(models.Model):
+    _name = "product.price.category"
+    _description = "ProductPriceCategory"
```
Comment (rousseldenis) [inline product_price_category/models/product_price_category.py]:
```suggestion
    _description = "Product Price Category"
```

File: product_price_category/models/product_pricelist_item.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        "to one price category. Keep empty otherwise.",
+    )
+
+    @api.onchange("applied_on")
+    def _onchange_applied_on_price_category(self):
+        """Reset the price_category_id value if applied_on
+        is not price_category
+        """
+        if self.applied_on != "2b_product_price_category":
+            self.price_category_id = False
```
Comment (rousseldenis) [inline product_price_category/models/product_pricelist_item.py]: Shouldn't be the 'price_category_id' field converted into computed field readonly False in order to remove onchange?

File: product_price_category/tests/test_sale.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
+# Copyright 2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestSale(TransactionCase):
+    def setUp(self):
```
Comment (rousseldenis) [inline product_price_category/tests/test_sale.py]: Could you convert this into class method ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by sbejaoui** [comment]
can you check @rousseldenis

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by lmignon — COMMENTED** [review]
LGTM (code + functional)

**Review by flotho — APPROVED** [review]
thanks for the change
LGTM

File: product_price_category/README.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+This module adds a field Price Category on Product Template
+and allow Pricelist to be applied on this field.
+
+Usage
+=====
+
+In Pricelist Form (Sales -> Configuration -> Pricelist), you can add or modify
+an item ("Manage Pricelist Items" access should be checked in the user settings)
+and select "Price Category" in "Applied on".
```
Comment (marielejeune) [inline product_price_category/README.rst:17]: "Advanced price rules (discounts, formulas)" pricelist option must be activated to enable this feature, maybe it is better to add this info in the README?

File: product_price_category/README.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+Product Price Category
+======================
+
+This module adds a field Price Category on Product Template
+and allow Pricelist to be applied on this field.
+
+Usage
+=====
+
+In Pricelist Form (Sales -> Configuration -> Pricelist), you can add or modify
```
Comment (marielejeune) [inline product_price_category/README.rst:15]: Since we need `sale_management` module to configure the items, shouldn't we depend on it in the manifest?

**Comment by rousseldenis** [comment]
@sbejaoui

File: product_price_category/README.rst (hunk 6 above)
Comment (rousseldenis) [reply product_price_category/README.rst:17]: @sbejaoui

File: product_price_category/README.rst (hunk 7 above)
Comment (rousseldenis) [reply product_price_category/README.rst:15]: No, but if we depends on menus defined there

**Comment by IT-Ideas** [comment]
Hello @sbejaoui I just pushed a fix commit as the name was not computed correctly when `applied_on` value is `2b_product_price_category`.

**Review by JordiMForgeFlow — APPROVED** [review]
Code + functional review 👍🏼

**Comment by rousseldenis** [comment]
All comments attended 

/ocabot merge nobump

**Pull request merged**